templates/nextjs/src/app/faucet/page.tsx

```diff
@@ -24,6 +29,25 @@ export default function Faucet() {
     }
   }, [wallet]);
 
+  useEffect(() => {
+    const interval = setInterval(async () => {
+      await refreshWalletBalance?.();
+    }, 500);
```

Comment:
Couple you up this to 5s to be safe? I kept getting rate limited when working against testnet with a 1s interval.